packages/shared/src/components/GrowthBookProvider.tsx

```diff
@@ -168,7 +168,7 @@ export const GrowthBookProvider = ({
 };
 
 export const useFeatureIsOn = (feature: Feature<JSONValue>): boolean =>
-  gbUseFeatureIsOn(feature.id);
+  gbUseFeatureIsOn(feature?.id);
```

Comment:
how can this happen since we have our features hardcoded in imports?

Reply:
This is always evaluated, but for referral there are [cases](https://github.com/dailydotdev/apps/pull/2335/files#diff-612332c695635127288cbed4bd774bd6653f2fafe99e39017fbab20ae44c618fR28-R30) where they not linked to a feature.
Let me know if you think other solutions might be better?
I can't dynamic render the hook though

```
const featureFlag = campaignFeatureFlagMap[campaignKey];
  const isCampaignEnabled = useFeatureIsOn(featureFlag);
```

Reply:
`gbUseFeatureIsOn` requires feature so not sure what happens when we pass undefined. If we do this we need to also make `feature` optional in our `useFeatureIsOn`.